File: A/Boy_or_Girl/Gurvir-Sandhar.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){

	//get user input	
	string input;
	cin >> input;
	
	//array which each index represents an ascii char
	string charList[256];
	
	//keeps track of number of distinct chars in the input string
	int numDistinct;
		
	//for each char in the input string, check if it is in charList
	//if not, then add it and increment numDistinct
	for(int i = 0; i < input.length(); i++){ 
		if(charList[input[i]] != "true"){
			charList[input[i]] = "true";
			numDistinct++;			
		}		
	}

	//check if numDistinct is odd
	if(numDistinct % 2 == 1) {
		cout << "IGNORE HIM!" << endl;
	}
	else
		cout << "CHAT WITH HER!" << endl;



}

File: A/Boy_or_Girl/Gurvir-Sandhar.js
//file can be ran by running: node Gurvir-Sandhar <input string>

//user input
let input = process.argv[2];

//holds distinct chars
let charList = {};

//keeps track of number of distinct chars
let numDistinct = 0;

//loop through input and check if currentChar is in charList
//else add currentChar to charList and increment numDistinct
for(let i = 0; i < input.length; i++) {
    let currentChar = input[i];
    if(!charList.hasOwnProperty(currentChar)) {
        charList[currentChar] = true;
        numDistinct++;
    }
}

//check if number of distinct chars is odd
if(numDistinct % 2 == 1) {
    console.log("IGNORE HIM!");
} else {
    console.log("CHAT WITH HER!");
}

